components/service/glean/src/test/java/mozilla/components/service/glean/storages/GenericScalarStorageEngineTest.kt

```diff
@@ -394,4 +419,73 @@ class GenericScalarStorageEngineTest {
             assertEquals(37, snapshot["telemetry.userLifetimeData"])
         }
     }
+
+    @Test
+    fun `metrics with 'ping' lifetime must be cleared when the ping is scheduled`() {
+        // We use block scopes to simulate restarting the application. We use the same
+        // context otherwise the test environment will use a different underlying file
+        // for the SharedPreferences.
+        run {
+            val storageEngine = MockScalarStorageEngine()
+            storageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+
+            val metric1 = GenericMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.User,
+                name = "userLifetimeData",
+                sendInPings = listOf("store1")
+            )
+
+            // Record a value with User lifetime
+            storageEngine.record(
+                metric1,
+                value = 37
+            )
+
+            val metric2 = GenericMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Ping,
+                name = "pingLifetimeData",
+                sendInPings = listOf("store1", "store2")
```

Comment:
I would also check that data is never cleared from "store2" in every `run` block below.